File: corrida-jegues/jegue.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.IOException;
import java.util.Map;
import java.util.TreeMap;

public class jegue {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Map<Integer, String> venc_t1 = new TreeMap<Integer, String>();
        Map<Integer, String> venc_t2 = new TreeMap<Integer, String>();
        Map<Integer, String> venc_chegada = new TreeMap<Integer, String>();

        while (true) {
            String str = br.readLine();
            if (str == null)
                break;
            else {
                String[] linha = str.split(" ");
                venc_t1.put(Integer.parseInt(linha[1]), linha[0]);
                venc_t2.put(Integer.parseInt(linha[2]), linha[0]);
                venc_chegada.put(Integer.parseInt(linha[3]), linha[0]);
            }
        }
        bw.write(telemetria(venc_t1,venc_t2,venc_chegada));
        bw.flush();
    }

    public static String telemetria(Map<Integer, String> t1, Map<Integer, String> t2, Map<Integer, String> cheg) {
        return String.format("%s%s%s", ordenar("T1", t1), ordenar("T2", t2), ordenar("CHEGADA", cheg));
    }

    public static String ordenar(String checkpoint, Map<Integer, String> map) {
        return String.format("%s %s %s %s\n", checkpoint, map.values().toArray()[0], map.values().toArray()[1],
                map.values().toArray()[2]);
    }
}

File: jegues/jegue.py
import collections
import sys

competidores = sys.stdin.readlines()

def telemetria(checkpoint, dados):
    dados = list(sorted(dados.items()))
    print('{} {} {} {}'.format(checkpoint, dados[0][1],dados[1][1],dados[2][1]))
    

venc_t1 = {}
venc_t2 = {}
venc_chegada = {}
for comp in competidores:
    nome, t1, t2, chegada = comp.rstrip('\n').split(' ')
    venc_t1[int(t1)] = nome
    venc_t2[int(t2)] = nome
    venc_chegada[int(chegada)] = nome

telemetria('T1', venc_t1)
telemetria('T2', venc_t2)
telemetria('CHEGADA', venc_chegada)